src/error-handling.cpp

```diff
@@ -49,28 +49,35 @@ namespace librealsense
          {
              try
              {
-
                  auto val = static_cast<uint8_t>(_option->query());
 
                  if (val != 0 && !_silenced)
                  {
                      auto strong = _notifications_processor.lock();
-                     if (strong) strong->raise_notification(_decoder->decode(val));
 
-                     val = static_cast<uint8_t>(_option->query());
-                     if (val != 0)
+                     // First reset the value in the FW.
+                     // Reading from last-error control is supposed to set it to zero in the
+                     // firmware If this is not happening there is some issue 
+                     // Note: if an error will be raised between the 2 queries, this will cause the error polling loop to stop
+                     auto reseted_val = static_cast<uint8_t>( _option->query() );
+                     if( reseted_val != 0 )
                      {
-                         // Reading from last-error control is supposed to set it to zero in the firmware
-                         // If this is not happening there is some issue
+                         std::string error_str = to_string()
+                                              << "Error polling loop is not behaving as expected! "
+                                                 "expecting value : 0 got : " << std::to_string( val )
+                                              << "\nShutting down error polling loop";
+                         LOG_ERROR( error_str );
                          notification postcondition_failed{
                              RS2_NOTIFICATION_CATEGORY_HARDWARE_ERROR,
                              0,
                              RS2_LOG_SEVERITY_WARN,
-                             "Error polling loop is not behaving as expected!\nThis can indicate an issue with camera firmware or the underlying OS..."
+                             error_str
                          };
-                         if (strong) strong->raise_notification(postcondition_failed);
+                         if( strong ) strong->raise_notification( postcondition_failed );
```

Comment:
I think we should LOG_DEBUG here, too, no? Do we show it any other way in the current logs?